app/controllers/devise_token_auth/confirmations_controller.rb

```diff
@@ -11,39 +25,21 @@ def show
 
         redirect_header_options = { account_confirmation_success: true }
 
-        if signed_in?(resource_name)
-          client_id, token = signed_in_resource.create_token
+        client_id, token = @resource.create_token
```

Comment:
Remove this as http://blog.plataformatec.com.br/2013/08/devise-3-1-now-with-more-secure-defaults/